### PR TITLE
Fixing race condition in channel test

### PR DESCRIPTION
**Pull request by hkaiser**
## Proposed Changes

  - fix a race in test channel_2916, the counter of pending tasks was relying on another thread being executed __after__ checking the value

## Any background context you want to provide?

- this was exposed by the recent patch fixing direct execution (#3104)

### CODE REVIEW

File: tests/regressions/lcos/CMakeLists.txt [hunk 1]
```diff
@@ -66,7 +66,7 @@ set(async_callback_non_deduced_context_PARAMETERS THREADS_PER_LOCALITY 4)
 set(broadcast_unwrap_future_2885_PARAMETERS LOCALITIES 2 THREADS_PER_LOCALITY 4)
 set(broadcast_wait_for_2822_PARAMETERS LOCALITIES 2 THREADS_PER_LOCALITY 4)
 set(channel_2916_FLAGS DEPENDENCIES iostreams_component)
-set(channel_2916_PARAMETERS LOCALITIES 2 THREADS_PER_LOCALITY 4)
+set(channel_2916_PARAMETERS THREADS_PER_LOCALITY 4)
```
Comment (sithhell) [inline tests/regressions/lcos/CMakeLists.txt]: why did you remove distributed testing here?

Comment (hkaiser) [reply tests/regressions/lcos/CMakeLists.txt]: Because the test itself does not do any distributed work.

Comment (sithhell) [reply tests/regressions/lcos/CMakeLists.txt]: Ah, yes sure...
